setup.py

```diff
@@ -69,54 +71,139 @@
+    usage: python setup.py deps_table_update
+    """
+    
+    description = 'build runtime dependency table'
+    user_options = [
+        # format: (long option, short option, description).
+        ('dep-table-update', None, 'updates src/transformers/dependency_versions_table.py'),
+    ]
+    def initialize_options(self): pass
+    def finalize_options(self): pass
+
+    def run(self):
+        target = "src/transformers/dependency_versions_table.py"
+        print(f"updating {target}")
+        with open(target, "w") as f:
+            entries = "\n".join([f'    "{k}": "{v}",' for k,v in deps.items()])
+            f.write("""# THIS FILE HAS BEEN AUTOGENERATED. To update:
+# 1. modify the `_deps` dict in setup.py
+# 2. run `make deps_table_update``
+deps = {
+""" + entries + "\n}\n")
+
+extras = {}
+
+extras["ja"] = deps_list("fugashi", "ipadic", "unidic_lite", "unidic")
+extras["sklearn"] = deps_list("scikit-learn")
+
+extras["tf"] = deps_list( "tensorflow", "onnxconverter-common", "keras2onnx")
+extras["tf-cpu"] = deps_list("tensorflow-cpu", "onnxconverter-common", "keras2onnx")
+
+extras["torch"] = deps_list("torch")
 
 if os.name == "nt":  # windows
-    extras["retrieval"] = ["datasets"] # faiss is not supported on windows
+    extras["retrieval"] = deps_list("datasets") # faiss is not supported on windows
     extras["flax"] = [] # jax is not supported on windows
 else:
-    extras["retrieval"] = ["faiss-cpu", "datasets"]
-    extras["flax"] = ["jaxlib==0.1.55", "jax>=0.2.0", "flax==0.2.2"]
+    extras["retrieval"] = deps_list("faiss-cpu", "datasets")
+    extras["flax"] = deps_list("jax", "jaxlib", "flax")
 
-extras["tokenizers"] = ["tokenizers==0.9.4"]
-extras["onnxruntime"] = ["onnxruntime>=1.4.0", "onnxruntime-tools>=1.4.2"]
-extras["modelcreation"] = ["cookiecutter==1.7.2"]
+extras["tokenizers"] = deps_list("tokenizers")
+extras["onnxruntime"] = deps_list("onnxruntime", "onnxruntime-tools")
+extras["modelcreation"] = deps_list("cookiecutter")
 
-extras["serving"] = ["pydantic", "uvicorn", "fastapi", "starlette"]
+extras["serving"] = deps_list("pydantic", "uvicorn", "fastapi", "starlette")
 
-extras["sentencepiece"] = ["sentencepiece==0.1.91", "protobuf"]
-extras["retrieval"] = ["faiss-cpu", "datasets"]
-extras["testing"] = ["pytest", "pytest-xdist", "timeout-decorator", "parameterized", "psutil"] + extras["retrieval"] + extras["modelcreation"]
-# sphinx-rtd-theme==0.5.0 introduced big changes in the style.
-extras["docs"] = ["recommonmark", "sphinx==3.2.1", "sphinx-markdown-tables", "sphinx-rtd-theme==0.4.3", "sphinx-copybutton"]
-extras["quality"] = ["black >= 20.8b1", "isort >= 5.5.4", "flake8 >= 3.8.3"]
+extras["sentencepiece"] = deps_list("sentencepiece", "protobuf")
+extras["retrieval"] = deps_list("faiss-cpu", "datasets")
+extras["testing"] = deps_list("pytest", "pytest-xdist", "timeout-decorator", "parameterized", "psutil") + extras["retrieval"] + extras["modelcreation"]
```

Comment:
This line is a bit too long, could you split it to 119-max?

Reply:
I re-run the whole thing through black/isort/flake8 - which of course made things look weird as usual but consistent in its own inconsistent way.